staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -268,6 +273,11 @@ func NewReflectorWithOptions(lw ListerWatcher, expectedType interface{}, store S
 	return r
 }
 
+func makeValidPromethusMetricName(in string) string {
+	// this isn't perfect, but it removes our common characters
+	return strings.NewReplacer("/", "_", ".", "_", "-", "_").Replace(in)
```

Comment:
instead of dropping specific bad characters, shouldn't this be inverted and replace any character not in the set of allowed characters for valid prometheus names?

Reply:
Yes, The current approach of replacing specific characters is less robust than enforcing the Prometheus metric naming rules directly. According to Prometheus documentation, metric names must match the regex `[a-zA-Z_:][a-zA-Z0-9_:]*. ` I'll adjust this code.

Reply:
@liggitt done